src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,105 @@
-export const PeopleFilters = () => {
+import { Link, useSearchParams } from 'react-router-dom';
+import { Person } from '../types/Person';
+import React, { useState, useEffect } from 'react';
+
+interface PeopleFiltersProps {
+  persons: Person[];
+  onFilterChange: (filter: 'all' | 'male' | 'female') => void;
+
+  onClick: (century: string | 'all' | null) => void;
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export const PeopleFilters: React.FC<PeopleFiltersProps> = ({
+  onFilterChange,
+  onClick,
+  onChange,
+}) => {
+  const [filter, setFilter] = useState<'all' | 'male' | 'female'>('all');
+  const [searchQuery, setSearchQuery] = useState<string>('');
+  const [searchParams, setSearchParams] = useSearchParams();
+  const centuryQuery = searchParams.getAll('century') || [];
```

Comment:
The `getAll` method returns an array, so `centuryQuery` will always be an array. The fallback to an empty array is unnecessary.